Make the defaults sane for Debian. (Do use mktemp!)

Index: dphys-config-20090605~curr/dphys-config
===================================================================
--- dphys-config-20090605~curr.orig/dphys-config	2009-06-05 23:08:52.000000000 +0200
+++ dphys-config-20090605~curr/dphys-config	2009-06-08 12:38:10.000000000 +0200
@@ -25,7 +25,7 @@
 # method for generating temp directory name for work files
 #   some prefer -$$- PIDs as no utility is needed, works everywhere
 #   some prefer mktemp as this prevents race conditions
-CONF_USE_MKTEMP=no
+CONF_USE_MKTEMP=yes
 
 # base directory for config files on server
 #   set to something that will produce an error message without site config
