Index: dphys-config/dphys-config
===================================================================
--- dphys-config.orig/dphys-config	2013-02-11 19:33:06.000000000 +0100
+++ dphys-config/dphys-config	2013-02-11 19:33:06.000000000 +0100
@@ -671,7 +671,7 @@
               # prevent non zero return from user given command from killing us
               set +e
               local SUBST
-              SUBST="`"${TICK}"`"
+              SUBST="`${TICK}`"
               RETVAL="$?"
               set -e
               if [ "${RETVAL}" != 0 ] ; then
