Enables dphys-config to report to hobbit servers

Index: dphys-config/dphys-config
===================================================================
--- dphys-config.orig/dphys-config	2011-08-18 17:06:07.000000000 +0200
+++ dphys-config/dphys-config	2011-08-18 17:09:03.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/dash
 # /usr/bin/dphys-config - install/update config files and trigger commands
 # author Neil Franklin, last modification 2010.01.21
 # copyright ETH Zuerich Physics Departement,
@@ -169,6 +169,79 @@
   . ~/."${PNAME}"
 fi
 
+# Setup hobbit support
+if [ -e /etc/default/hobbit-client ] ; then
+  . /etc/default/hobbit-client
+fi
+if [ -n "${HOBBITSERVERS}" -a -x '/usr/lib/hobbit/client/bin/bb' ] ; then
+    HOBBITCOLOR=green
+    HOBBITMESSAGE=""
+
+    color_line() {
+	LOCALCOLOR="$1"
+	shift
+	SYSLOGLEVEL="$1"
+	shift
+	HOBBITMESSAGE="${HOBBITMESSAGE}&${LOCALCOLOR} $*
+"
+	if [ "${HOBBITCOLOR}" = 'green' ]; then
+	    ${CMD_DEBUG_PRINT} "Setting color to ${LOCALCOLOR}"
+	    HOBBITCOLOR="${LOCALCOLOR}"
+	elif [ "${HOBBITCOLOR}" = 'yellow' -a "${LOCALCOLOR}" = 'red' ]; then
+	    ${CMD_DEBUG_PRINT} "Setting color to ${LOCALCOLOR}"
+	    HOBBITCOLOR="${LOCALCOLOR}"
+	fi
+	${CMD_DEBUG_PRINT} "color_line: HOBBITCOLOR is ${HOBBITCOLOR}"
+	logger -t "${BASENAME}" -p user."${SYSLOGLEVEL}" "$*"
+    }
+
+    hobbit_line() {
+	HOBBITMESSAGE="${HOBBITMESSAGE}$*
+"
+	${CMD_DEBUG_PRINT} "hobbit_line: HOBBITCOLOR is ${HOBBITCOLOR}"
+	logger -t "${BASENAME}" -p user.info "$*"
+    }
+
+    hobbit_printonly_line() {
+	HOBBITMESSAGE="${HOBBITMESSAGE}$*
+"
+	${CMD_DEBUG_PRINT} "hobbit_printonly_line: HOBBITCOLOR is ${HOBBITCOLOR}"
+	"${CMD_INFO}" "$*"
+    }
+
+    hobbit_send() {
+	${CMD_DEBUG_PRINT} "Sending the following status report (${HOBBITCOLOR}) to ${HOBBITSERVERS}:"
+	${CMD_DEBUG_SLEEP}
+	if [ "${HOBBITCOLOR}" = 'green' ]; then
+	    HOBBITOK="OK"
+	else
+	    HOBBITOK="NOT OK"
+	fi
+	DATE=`date`
+	HOBBITSTATUS="status+${HOBBITVALIDITY:-2160} ${CLIENTHOSTNAME:-${CONF_CONFNAME}}.${HOBBIT_TESTNAME:-dpc} ${HOBBITCOLOR} ${DATE} dphys-config ${HOBBITOK}
+
+${HOBBITMESSAGE}"
+	${CMD_DEBUG_PRINT} "${HOBBITSTATUS}"
+	for hobbitserver in ${HOBBITSERVERS}; do
+	    /usr/lib/hobbit/client/bin/bb $hobbitserver "${HOBBITSTATUS}"
+	done
+    }
+
+    # something within the system that user does not expect, give up
+    CMD_LOG_FATAL="color_line red error $0: FATAL:"
+    # something from users input, user will correct this and continue
+    CMD_LOG_ERROR="color_line red error $0: ERROR:"
+    # something we can continue with, but may be wrong, and user may not suspect it
+    CMD_LOG_WARNING="color_line yellow warning $0: WARNING:"
+    # something most likely not wrong, but tell user for the odd case it is wrong
+    CMD_LOG_NOTE="color_line green notice $0: NOTE:"
+    # normal stuff users expect, so no marking
+    CMD_LOG_INFO="hobbit_line"
+    CMD_INFO_PRINT="hobbit_printonly_line"
+
+    # Send hobbit report on exit or error
+    trap hobbit_send 0
+fi
 
 # --- control variable output
 
@@ -715,6 +788,8 @@
 ${CMD_INFO_PRINT} "processing individual config files from the list ..."
 ${CMD_DEBUG_SLEEP}
 
+trap - 0
+
 # cut get rid of comments and grep empty lines (may be emptied comment lines)
 #   and allow user to select/filter subset of lines to be processed
 #   IFS= to avoid read doing separator processing (mangles spaces)
@@ -722,6 +797,8 @@
 cut -f 1 -d '#' "${CONFLIST_PLACE}" | grep -v '^ *$' | \
     grep "${CONF_LINEFILTER}" | while IFS= read -r LINE ; do
 
+    trap hobbit_send 0
+
   ${CMD_DEBUG_PRINT} "LINE=${LINE}"
   ${CMD_DEBUG_WAIT}
 
