Author: Claude Becker <becker@phys.ethz.ch>
Author: Axel Beckert <beckert@phys.ethz.ch>
Description: Adds options -i and -d for interctive and diff mode.

Index: dphys-config/dphys-config
===================================================================
--- dphys-config.orig/dphys-config	2012-08-14 14:32:20.000000000 +0200
+++ dphys-config/dphys-config	2012-08-14 14:32:20.000000000 +0200
@@ -136,6 +136,18 @@
 CMD_INFO_PRINT="${CMD_INFO}"
 CMD_VERBOSE_PRINT=true
 
+# command to use for viewing diffs in interactive mode
+if [ -x /usr/bin/colordiff ] ; then
+  CMD_DIFF='colordiff -w -u'
+else
+  CMD_DIFF='diff -w -u'
+fi
+
+if [ -x /usr/bin/dwdiff ] ; then
+  CMD_WDIFF='dwdiff -c'
+else
+  CMD_WDIFF='wdiff'
+fi
 
 # set debug option controllable stuff
 if [ "${DEBUG_SLEEP}" = yes ] ; then
@@ -210,6 +222,33 @@
         shift 1
         ;;
 
+      i)
+        # interactive mode: prompt for action before updating file
+        INTERACTIVE_MODE=true
+        ;;
+
+      d)
+        # diff-only mode: show diff of changed files before installing them
+        DIFF_ONLY_MODE=true
+        ;;
+
+      D)
+        # diff+install mode: show diff of changed files before installing them
+        DIFF_INSTALL_MODE=true
+        ;;
+
+      w)
+        # diff+install mode: show diff of changed files before installing them
+        DIFF_ONLY_MODE=true
+        WDIFF_MODE=true
+        ;;
+
+      W)
+        # diff+install mode: show diff of changed files before installing them
+        DIFF_INSTALL_MODE=true
+        WDIFF_MODE=true
+        ;;
+
       q)
         # quiet: don't inform user what we are doing
         CMD_INFO_PRINT=true
@@ -243,6 +282,11 @@
 options:
   -c         configname: use this set of config files instead of hostname set
   -f filter  filter: only process lines matching this grep regexp
+  -i         interactive mode: prompt for action before updating file
+  -d         diff-only mode: show diff of changed file but don't install them
+  -D         diff+install mode: show diff of changed files before installing them
+  -w         wdiff-only mode: show word-diff of changed file but don't install them
+  -W         wdiff+install mode: show word-diff of changed files before installing them
   -q         quiet: don't give user an running report of what we are doing
   -v         verbose: give user detailed running report
   -D         Debug: activate an debug option, see source for operation
@@ -731,6 +775,90 @@
 
 }
 
+# --- install new config file
+
+install_config_file () {
+  ${CMD_INFO_PRINT} "updating old or installing new config" \
+      "${CONFFILE_NAME} as ${CONFFILE_PLACE} ..."
+  ${CMD_DEBUG_SLEEP}
+
+  cp -pf "${CONFFILE}" "${CONFFILE_PLACE}"
+
+  ${CMD_DEBUG_PRINT} "`ls -al "${CONFFILE_PLACE}"`"
+  ${CMD_DEBUG_WAIT}
+
+
+  # run the facultative postinstall command trigger stuff
+  if [ "`echo "${CMDTRIG}" | tr -d ' '`" != "" ] ; then
+
+    ${CMD_INFO_PRINT} "triggering postinstall command \"${CMDTRIG}\" ..."
+    ${CMD_DEBUG_SLEEP}
+
+    # prevent non zero return from user given command from killing us
+    set +e
+    sh -c "${CMDTRIG}"
+    RETVAL="$?"
+    set -e
+    if [ "${RETVAL}" != 0 ] ; then
+      ${CMD_WARNING} "command returned non zero: ${RETVAL}" >&2
+      ${CMD_LOG_WARNING} "command returned non zero: ${RETVAL}"
+    fi
+
+    ${CMD_DEBUG_PRINT} "`ls -al "${CONFFILE_PLACE}"`"
+    ${CMD_DEBUG_WAIT}
+
+  fi
+
+  return 0
+}
+
+# --- remove existing config file
+
+remove_config_file () {
+  ${CMD_INFO_PRINT} "removing old config file ${CONFFILE_PLACE} ..."
+  ${CMD_DEBUG_SLEEP}
+
+  # test for only directory name (ends in /), possibly dangerous leftover
+  if [ "`echo "${CONFFILE_PLACE}" | rev | cut -c 1`" = "/" ] ; then
+
+    ${CMD_WARNING} "would remove entire directory ${CONFFILE_PLACE}," \
+        "may be partially edited list file entry, DANGEROUS, leaving it," \
+        "add the files name after the / to delete an single file," \
+        "remove / after dir name to really delete entire directory" >&2
+    ${CMD_LOG_WARNING} "would remove entire directory ${CONFFILE_PLACE}," \
+        "may be partially edited list file entry, DANGEROUS, leaving it," \
+        "add the files name after the / to delete an single file," \
+        "remove / after dir name to really delete entire directory"
+
+  else
+
+    # run the facultative preremove command trigger stuff
+    if [ "`echo "${CMDTRIG}" | tr -d ' '`" != "" ] ; then
+
+      ${CMD_INFO_PRINT} "triggering preremove command \"${CMDTRIG}\" ..."
+      ${CMD_DEBUG_SLEEP}
+
+      set +e
+      sh -c "${CMDTRIG}"
+      RETVAL="$?"
+      set -e
+      if [ "${RETVAL}" != 0 ] ; then
+        ${CMD_WARNING} "command returned non zero: ${RETVAL}" >&2
+        ${CMD_LOG_WARNING} "command returned non zero: ${RETVAL}"
+      fi
+
+    fi
+
+    ${CMD_INFO_PRINT} "removing config file ${CONFFILE_PLACE} ..."
+    ${CMD_DEBUG_SLEEP}
+
+    ${CMD_DEBUG_PRINT} "`ls -ald "${CONFFILE_PLACE}"`"
+    ${CMD_DEBUG_WAIT}
+
+    rm -rf "${CONFFILE_PLACE}"
+
+  fi
+}
 
 # --- download config file list
 
@@ -915,36 +1043,75 @@
     # only if file has changed do we do anything, as for config list
     if ! diff "${CONFFILE_PLACE}" "${CONFFILE}" > /dev/null ; then
 
-      ${CMD_INFO_PRINT} "updating old or installing new config" \
-          "${CONFFILE_NAME} as ${CONFFILE_PLACE} ..."
-      ${CMD_DEBUG_SLEEP}
-
-      cp -pf "${CONFFILE}" "${CONFFILE_PLACE}"
-
-      ${CMD_DEBUG_PRINT} "`ls -al "${CONFFILE_PLACE}"`"
-      ${CMD_DEBUG_WAIT}
-
-
-      # run the facultative postinstall command trigger stuff
-      if [ "`echo "${CMDTRIG}" | tr -d ' '`" != "" ] ; then
-
-        ${CMD_INFO_PRINT} "triggering postinstall command \"${CMDTRIG}\" ..."
-        ${CMD_DEBUG_SLEEP}
-
-        # prevent non zero return from user given command from killing us
-        set +e
-        sh -c "${CMDTRIG}"
-        RETVAL="$?"
-        set -e
-        if [ "${RETVAL}" != 0 ] ; then
-          ${CMD_WARNING} "command returned non zero: ${RETVAL}" >&2
-          ${CMD_LOG_WARNING} "command returned non zero: ${RETVAL}"
-        fi
+      ACTION_EXECUTED=false
 
-        ${CMD_DEBUG_PRINT} "`ls -al "${CONFFILE_PLACE}"`"
-        ${CMD_DEBUG_WAIT}
+      while [ $ACTION_EXECUTED != true ] ; do
+          if [ $INTERACTIVE_MODE ] ; then
+              # prompt user for action
+              echo ""
+              printf "Found new version for ${CONFFILE_PLACE}\n"
+              printf "[i]nstall [d]iff [D]iff+install [w]diff [W]diff+install [s]kip "
+              ACTION=$(bash --noprofile --norc -c 'read -n 1 ACTION < /dev/tty; echo $ACTION')
+              echo ""
+          elif [ $DIFF_ONLY_MODE ] ; then
+              # default action is to diff-only
+	       if [ $WDIFF_MODE ]; then
+	           ACTION='w'
+	       else
+                  ACTION='d'
+	       fi
+          elif [ $DIFF_INSTALL_MODE ] ; then
+              # default action is to diff+install
+	       if [ $WDIFF_MODE ]; then
+	           ACTION='W'
+	       else
+                  ACTION='D'
+	       fi
+          else
+              # default action is to install
+              ACTION='i'
+          fi
 
-      fi
+          case $ACTION in
+              s|n)  # skip file
+                  ACTION_EXECUTED=true
+                  echo "Skipping this file and keeping currently installed version."
+                  ;;
+              i|y)  # install file
+                  ACTION_EXECUTED=true
+                  install_config_file
+                  ;;
+              d)  # diff file
+		  if [ $DIFF_ONLY_MODE ]; then
+		      ACTION_EXECUTED=true
+		  fi
+                  ${CMD_DIFF} ${CONFFILE_PLACE} ${CONFFILE} || true
+                  ;;
+              D)  # diff + install file
+                  ACTION_EXECUTED=true
+                  ${CMD_DIFF} ${CONFFILE_PLACE} ${CONFFILE} || true
+                  install_config_file
+                  ;;
+              w)  # wdiff file
+		  if [ $DIFF_ONLY_MODE ]; then
+		      ACTION_EXECUTED=true
+		  fi
+                  ${CMD_WDIFF} ${CONFFILE_PLACE} ${CONFFILE} || true
+                  ;;
+              W)  # wdiff + install file
+                  ACTION_EXECUTED=true
+                  ${CMD_WDIFF} ${CONFFILE_PLACE} ${CONFFILE} || true
+                  install_config_file
+                  ;;
+              q)  # quit
+	       	   echo "Skipping this and all other pending files."
+                  exit 42
+                  ;;
+              *)
+                  echo "Unknown action '$ACTION'."
+                  ;;
+          esac
+      done
 
     fi
 
@@ -955,49 +1122,39 @@
     # only if file still exists do we do anything
     if [ -e "${CONFFILE_PLACE}" ] ; then
 
-      ${CMD_INFO_PRINT} "removing old config file ${CONFFILE_PLACE} ..."
-      ${CMD_DEBUG_SLEEP}
-
-      # test for only directory name (ends in /), possibly dangerous leftover
-      if [ "`echo "${CONFFILE_PLACE}" | rev | cut -c 1`" = "/" ] ; then
-
-        ${CMD_WARNING} "would remove entire directory ${CONFFILE_PLACE}," \
-            "may be partially edited list file entry, DANGEROUS, leaving it," \
-            "add the files name after the / to delete an single file," \
-            "remove / after dir name to really delete entire directory" >&2
-        ${CMD_LOG_WARNING} "would remove entire directory ${CONFFILE_PLACE}," \
-            "may be partially edited list file entry, DANGEROUS, leaving it," \
-            "add the files name after the / to delete an single file," \
-            "remove / after dir name to really delete entire directory"
+      ACTION_EXECUTED=false
 
-      else
-
-        # run the facultative preremove command trigger stuff
-        if [ "`echo "${CMDTRIG}" | tr -d ' '`" != "" ] ; then
-
-          ${CMD_INFO_PRINT} "triggering preremove command \"${CMDTRIG}\" ..."
-          ${CMD_DEBUG_SLEEP}
-
-          set +e
-          sh -c "${CMDTRIG}"
-          RETVAL="$?"
-          set -e
-          if [ "${RETVAL}" != 0 ] ; then
-            ${CMD_WARNING} "command returned non zero: ${RETVAL}" >&2
-            ${CMD_LOG_WARNING} "command returned non zero: ${RETVAL}"
+      while [ $ACTION_EXECUTED != true ] ; do
+          if [ $INTERACTIVE_MODE ] ; then
+              # prompt user for action
+              echo ""
+              printf "Found config file ${CONFFILE_PLACE} to be deleted\n"
+              printf "[r]emove [v]iew [s]kip "
+              ACTION=$(bash --noprofile --norc -c 'read -n 1 ACTION < /dev/tty; echo $ACTION')
+              echo ""
+          else
+              # default action is to remove
+              ACTION='r'
           fi
 
-        fi
+          case $ACTION in
+              s|n)  # skip file
+                  ACTION_EXECUTED=true
+                  echo "Skipping this file and keeping currently installed version."
+                  ;;
+              r|y)  # remove file
+                  ACTION_EXECUTED=true
+                  remove_config_file
+                  ;;
+              v)  # view file
+                  ${CMD_DIFF} ${CONFFILE_PLACE} /dev/null || true
+                  ;;
+              *)
+                  echo "Unknown action '$ACTION'."
+                  ;;
+          esac
+      done
 
-        ${CMD_INFO_PRINT} "removing config file ${CONFFILE_PLACE} ..."
-        ${CMD_DEBUG_SLEEP}
-
-        ${CMD_DEBUG_PRINT} "`ls -ald "${CONFFILE_PLACE}"`"
-        ${CMD_DEBUG_WAIT}
-
-        rm -rf "${CONFFILE_PLACE}"
-
-      fi
 
     fi
 
